src/lib/dom.js

```diff
@@ -91,11 +93,66 @@ function deleteRelatedStyleRule(uid) {
     if(style) removeElement(style);
 }
 
+function getFullTransformMatrix(element) {
+    var allElements = getElementAndAncestors(element);
+    // the identity matrix
+    var out = [
+        1, 0, 0, 0,
+        0, 1, 0, 0,
+        0, 0, 1, 0,
+        0, 0, 0, 1
+    ];
+    allElements.forEach(function(e) {
+        var t = getElementTransformMatrix(e);
+        if(t) {
+            var m = matrix.convertCssMatrix(t);
+            out = mat4X4.multiply(out, out, m);
+        }
+    });
+    return out;
+}
+
+/**
+ * extracts and parses the 2d css style transform matrix from some element
+ */
+function getElementTransformMatrix(element) {
+    var style = window.getComputedStyle(element, null);
+    var transform = (
+      style.getPropertyValue('-webkit-transform') ||
+      style.getPropertyValue('-moz-transform') ||
+      style.getPropertyValue('-ms-transform') ||
+      style.getPropertyValue('-o-transform') ||
+      style.getPropertyValue('transform')
+    );
+
+    if(transform === 'none') return null;
+    // the slice is because the transform string returns eg "matrix(0.5, 0, 1, 0, 1, 1)"
+    return transform.slice(7, -1).split(',').map(function(n) {return +n;});
```

Comment:
Do you always get the `matrix` form? There are lots of other options listed in https://developer.mozilla.org/en-US/docs/Web/CSS/transform

Minor: `.map(function(n) {return +n;})` can be `.map(Number)`

Reply:
If you do something like element.style.transform, you can get any of the options listed there. But if you do it via window.getComputedStyle, I believe it always gives you a matrix of that form.

Reply:
Good to know! Is that documented somewhere? If not we will want to test it in all supported browsers.

Reply:
Hm, I've been unable to find a clear bit of documentation to confirm this, just lots of uses of it on stackoverflow. I think it's legit but it definitely needs testing.

Reply:
I've found https://www.w3.org/TR/css-transforms-1/#serialization-of-the-computed-value

> A <transform-list> for the computed value is serialized to one <matrix()> function ...

and https://zellwk.com/blog/css-translate-values-in-javascript/

> Browsers turn transform values into 2d or 3d matrices depending on what transformations you applied.

Still seems like we should test this in IE11 at least, but modern browsers seem to be consistent in this.